applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/element_finite_difference_utility.cpp

```diff
@@ -77,11 +77,8 @@ namespace Kratos
     {
         KRATOS_TRY;
 
-        if( rDesignVariable == SHAPE_X || rDesignVariable == SHAPE_Y || rDesignVariable == SHAPE_Z )
+        if( rDesignVariable == SHAPE_SENSITIVITY_X || rDesignVariable == SHAPE_SENSITIVITY_Y || rDesignVariable == SHAPE_SENSITIVITY_Z )
         {
-            KRATOS_WARNING_IF("ElementFiniteDifferenceUtility::CalculateRightHandSideDerivative", OpenMPUtils::IsInParallel() != 0)
-                << "The call of this non omp-parallelized function within a parallel section should be avoided for efficiency reasons!" << std::endl;
-
             #pragma omp critical
```

Comment:
I think this #pragma omp critical has shown to not solve the thread safety problem and it would make sense to remove it here. It kind of gives the impression that this function could be called in an omp loop which it can't as you mentioned.

Reply:
:+1: done